sbt-scalajs-bundler/src/main/scala/scalajsbundler/util/ScalaJSOutputAnalyzer.scala

```diff
@@ -54,4 +54,39 @@ object ScalaJSOutputAnalyzer {
       .distinct
   }
 
+  /**
+    * @return The list of ES modules exported by a Scala.js project
+    * @param linkerConfig Configuration of the Scala.js linker
+    * @param linker Scala.js linker
+    * @param irFiles Scala.js IR files
+    * @param moduleInitializers Scala.js module initializers
+    * @param logger Logger
+    */
+  def findTopLevelExports(
+                           linkerConfig: StandardLinker.Config,
+                           linker: ClearableLinker,
+                           irFiles: Seq[VirtualScalaJSIRFile],
+                           moduleInitializers: Seq[ModuleInitializer],
+                           logger: Logger
+                         ): List[String] = {
+    require(linkerConfig.moduleKind == ModuleKind.CommonJSModule,
+      s"linkerConfig.moduleKind was ${linkerConfig.moduleKind}")
+    val symbolRequirements = {
+      val backend = new BasicLinkerBackend(linkerConfig.semantics,
+        linkerConfig.outputMode,
+        linkerConfig.moduleKind,
+        linkerConfig.sourceMap,
+        LinkerBackend.Config())
+      backend.symbolRequirements
+    }
+    val linkingUnit =
+      linker.linkUnit(irFiles,
```

Comment:
I've factored out getting the LinkingUnit from pulling stuff out of it. Would it be possible for us to hook into the LinkingUnit that was created by fastOptJS or is that not feasible?